airflow/api_fastapi/execution_api/routes/task_instances.py

```diff
@@ -122,6 +124,53 @@ def ti_update_state(
         )
     elif isinstance(ti_patch_payload, TITerminalStatePayload):
         query = TI.duration_expression_update(ti_patch_payload.end_date, query, session.bind)
+    elif isinstance(ti_patch_payload, TIDeferredStatePayload):
+        trigger_row = Trigger(
+            classpath=ti_patch_payload.classpath,
+            kwargs=ti_patch_payload.kwargs,
+            created_date=ti_patch_payload.created_date,
+        )
+        session.add(trigger_row)
+        session.flush()
```

Comment:
Shouldn't need it

Reply:
Will handle in next commit. Keeping open for now

Reply:
Actually we need this part. For deferred, we execute 2 queries, one is to update the trigger table, handled by this `session.flush()` and other to update the TI table, handled by the session.execute() below. We should keep this to take care of the `Trigger` table update. I dont want to extend the 
```
    try:
        result = session.execute(query)
        log.info("TI %s state updated: %s row(s) affected", ti_id_str, result.rowcount)
```

Part to more than one query for a one-off case. WDYT?

Reply:
Where is this query?

Reply:
Yeah but why do you need `session.flush`? The session will already be commit'ted and handled via the `get_session`

https://github.com/apache/airflow/blob/4d3140dba5976acb38c3dffd4fd963514f8da634/airflow/api_fastapi/execution_api/routes/task_instances.py#L67

Reply:
On checking further, I realised that it is not needed. The get_session context manager will handle it. The reason it was added was for the unit test portion.

After adding a session.commit to the unit test after calling the patch ideally represents the API and it works as expected.

```
            payload = {
                "state": "deferred",
                "classpath": "my-class-path",
                "kwargs": {"key": "value"},
                "next_method": "execute_callback",
                "timeout": "P1D",
            }

            response = client.patch(f"/execution/task-instances/{ti.id}/state", json=payload)

            assert response.status_code == 204
            assert response.text == ""

            session.commit()
            session.expire_all()
```